File: 1696-jump-game-vi/1696-jump-game-vi.java
class Solution {
    public int maxResult(int[] nums, int k) {
        PriorityQueue<Integer>pq=new PriorityQueue<Integer>();
        PriorityQueue<Integer>temp=new PriorityQueue<Integer>();
        int n=nums.length;
        pq.add(-nums[0]);
        
        for(int i=1,j=0;i<n;i++){
            int curr=(-pq.peek()+nums[i]);
            pq.add(-curr);
            nums[i]=curr;
            if(i-j==k){
                temp.add(-nums[j]);
                while(temp.size()>0 && pq.size()>0 && temp.peek().intValue()==pq.peek().intValue()){
                    temp.remove();
                    pq.remove();
                }
                j++;
            }
        }
        
        return nums[n-1];
    }
}

File: 1696-jump-game-vi/1696-jump-game-vi.cpp
class Solution {
public:
    int maxResult(vector<int>& nums, int k) {
        deque<int>q{0};
        
        for(int i=1,n=nums.size();i<n;i++){
            
            nums[i]+=nums[q.front()];
            
            while(q.size() && nums[q.back()]<nums[i])
                q.pop_back();
            
            q.push_back(i);
            
            if(i-q.front()==k){
                q.pop_front();
            }
        }
        
        return nums.back();
    }
};